src/lib/hooks/useTokenList/querying.ts

```diff
@@ -10,13 +11,25 @@ import { tokenComparator, useSortTokensByQuery } from './sorting'
 export function useQueryTokens(query: string, tokens: WrappedTokenInfo[]) {
   const { account } = useActiveWeb3React()
   const balances = useTokenBalances(account, tokens)
-  const sortedTokens = useMemo(() => [...tokens.sort(tokenComparator.bind(null, balances))], [balances, tokens])
+  const sortedTokens = useMemo(
+    () => [
+      // spread to create a new memo, because sort is in-place and returns referentially equivalent
+      ...tokens.sort(tokenComparator.bind(null, balances)),
+    ],
```

Comment:
nit: prefer `Array.from()`

Reply:
also, this only creates a new array _after_ sorting in place, which might muck up the data structure where it is generated. good call!